src/clients/AcrossAPIClient.ts

```diff
@@ -73,11 +75,15 @@ export class AcrossApiClient {
         const destinationChains = Object.keys(l2TokenAddresses)
           .map((chainId) => Number(chainId))
           .filter((chainId) => {
-            return (
-              chainId !== CHAIN_IDs.MAINNET &&
-              mainnetSpokePoolClient.isDepositRouteEnabled(l1Token, chainId) &&
-              Object.keys(this.spokePoolClients).includes(chainId.toString())
-            );
+            try {
+              // Verify that a token mapping exists on the destination chain.
+              hubPoolClient.getL2TokenForL1TokenAtBlock(l1Token, chainId); // throws if not found.
+              return (
+                chainId !== hubPoolClient.chainId && Object.keys(this.spokePoolClients).includes(chainId.toString())
```

Comment:
Thoughts on removing `Object.keys` to reduce the O(N) conversion?

```suggestion
                chainId !== hubPoolClient.chainId && isDefined(this.spokePoolClients[chainId])
```

Reply:
Good call; though that code was actually from a preceding PR that was refactored before merge; it's not part of the current change anymore.